### PR TITLE
feat: add config code entry screen to dashboard

### PR DESCRIPTION
**Pull request by utkuozdemir**
Implement a screen for entering/managing the config `${code}` variable.

Enable this screen only when the platform is `metal` and there is a `${code}` variable in the `talos.config` kernel cmdline URL query.

Additionally, remove the "Delete" button and its functionality from the network config screen to avoid users accidentally deleting PlatformNetworkConfig parts that are not managed by the dashboard.

Add some tests for the form data parsing on the network config screen.

Closes [#6993](https://github.com/siderolabs/talos/issues/6993).

Preview:

configcode

### CODE REVIEW

File: internal/app/dashboard/main.go [hunk 1]
```diff
@@ -51,7 +54,30 @@ func dashboardMain() error {
...
+
+		if showConfigCodeTab() {
+			screens = append(screens, dashboard.ScreenConfigCode)
+		}
 	}
 
 	return dashboard.Run(adminCtx, c, dashboard.WithAllowExitKeys(false), dashboard.WithScreens(screens...))
 }
+
+func showConfigCodeTab() bool {
```
Comment (smira) [inline internal/app/dashboard/main.go:66]: :heart:

File: internal/app/machined/pkg/runtime/v1alpha1/platform/metal/url/variable.go [hunk 2]
```diff
@@ -28,31 +28,29 @@ type Variable struct {
...
-			Key:   constants.HostnameKey,
-			Value: HostnameValue(),
-		},
-		{
-			Key:   constants.CodeKey,
-			Value: CodeValue(),
-		},
-	}
+// AllVariables is a map of all supported variables.
+var AllVariables = map[string]*Variable{
```
Comment (smira) [inline internal/app/machined/pkg/runtime/v1alpha1/platform/metal/url/variable.go:32]: as `Variable` embeds `Value` which contains matched value, we should always generate a fresh (empty) list here, so this should be still a `func()`

**Review by frezbo — APPROVED** [review]

File: internal/app/machined/pkg/runtime/v1alpha1/platform/metal/url/variable.go (hunk 2 above)
Comment (utkuozdemir) [reply internal/app/machined/pkg/runtime/v1alpha1/platform/metal/url/variable.go]: got it, done.

**Comment by utkuozdemir** [comment]
> Since we have plans on making this much more generic and allowing for other variables besides `code` I feel the tab should be named something like "Config URL".

Changed it to be Config URL.

**Pull request merged**